src/main/java/com/labsynch/labseer/api/ApiSaltController.java

```diff
@@ -44,18 +73,202 @@ public ResponseEntity<String> loadSalts(
 		headers.add("Access-Control-Allow-Headers", "Content-Type");
+		Salt salt = Salt.fromJsonToSalt(json);
+		salt.setAbbrev(salt.getAbbrev().trim());
+		salt.setName(salt.getName().trim());
+		ArrayList<ErrorMessage> errors = new ArrayList<ErrorMessage>();
+		boolean validSalt = true;
+		List<Salt> saltsByName = Salt.findSaltsByNameEquals(salt.getName()).getResultList();
+		if (saltsByName.size() > 0) {
+			logger.error("Number of salts found: " + saltsByName.size());
+			validSalt = false;
+			ErrorMessage error = new ErrorMessage();
+			error.setLevel("error");
+			error.setMessage("Duplicate salt name. Another salt exist with the same name.");
+			errors.add(error);
+		}
+		List<Salt> saltsByAbbrev = Salt.findSaltsByAbbrevEquals(salt.getAbbrev()).getResultList();
+		if (saltsByAbbrev.size() > 0) {
+			logger.error("Number of salts found: " + saltsByAbbrev.size());
+			validSalt = false;
+			ErrorMessage error = new ErrorMessage();
+			error.setLevel("error");
+			error.setMessage("Duplicate salt abbreviation. Another salt exist with the same abbreviation.");
+			errors.add(error);
+		}
+		if (validSalt) {
+			try{
+				salt = saltStructureService.saveStructure(salt);
+			} catch (CmpdRegMolFormatException e) {
+				logger.error("Error saving salt: " + e.getMessage());
+				validSalt = false;
+				ErrorMessage error = new ErrorMessage();
+				error.setLevel("error");
+				error.setMessage("Error saving salt: " + e.getMessage());
+				errors.add(error);
+				return new ResponseEntity<String>(ErrorMessage.toJsonArray(errors), headers, HttpStatus.BAD_REQUEST);
+			}
+		}
+		if (salt.getCdId() == -1) {
+			ErrorMessage error = new ErrorMessage();
+			error.setLevel("error");
+			error.setMessage("Bad molformat. Please fix the molfile: " + salt.getMolStructure());
+			errors.add(error);
+			return new ResponseEntity<String>(ErrorMessage.toJsonArray(errors), headers, HttpStatus.BAD_REQUEST);
+		} else if (salt.getCdId() > 0) {
+			salt.persist();
+			return showJson(salt.getId());
+		} else {
+			ErrorMessage error = new ErrorMessage();
+			error.setLevel("warning");
+			error.setMessage("Duplicate salt found. Please select existing salt.");
+			errors.add(error);
+			return new ResponseEntity<String>(ErrorMessage.toJsonArray(errors), headers, HttpStatus.CONFLICT);
+		}
+	}
+
+	// @RequestMapping(method = RequestMethod.OPTIONS)
+	// public ResponseEntity<String> getSaltOptions() {
+	// 	HttpHeaders headers = new HttpHeaders();
+	// 	headers.add("Content-Type", "application/text, text/html");
+	// 	headers.add("Access-Control-Allow-Headers", "Content-Type");
+	// 	headers.add("Access-Control-Allow-Origin", "*");
+	// 	headers.add("Access-Control-Max-Age", "86400");
+	// 	return new ResponseEntity<String>(headers, HttpStatus.OK);
+	// }
+
+	@RequestMapping(method = RequestMethod.PUT)
+	public String update(@Valid Salt salt, BindingResult bindingResult, Model uiModel, HttpServletRequest httpServletRequest) {
```

Comment:
Not something you need to address here, but I'm pretty certain requests like this are only for the Roo admin GUI. So when you go to remove that in a later PR, looking for this `HttpServletRequest` argument will be a good way to detect routes to remove.

Reply:
Thanks, yeah, I saw a number of these and wanted to leave it in because I didn't want to cross the "rename" threshold for git.  I will remove them later.